code/game/jobs/job_controller.dm

```diff
@@ -365,6 +348,23 @@ var/global/datum/controller/occupations/job_master
 			unassigned -= player
 	return 1
 
+/datum/controller/occupations/proc/TryAssignJob(var/mob/new_player/player, var/level, var/datum/job/job)
+	if(!job)
+		return FALSE
+	if(jobban_isbanned(player, job.title))
+		Debug("DO isbanned failed, Player: [player], Job:[job.title]")
```

Comment:
Will you keep the debug uncommented?

Reply:
That is as it was (see deletions for this file), so yes. All I'm doing here is moving this function into its own proc so that I can call it directly.